packages/gatsby/src/query/queue.js

```diff
@@ -97,11 +98,25 @@ const processBatch = async (queue, jobs, activity) => {
       queue.on(`task_finish`, () => activity.tick())
     }
 
+    const gc = () => {
+      queue.removeAllListeners(`task_failed`)
+      queue.removeAllListeners(`drain`)
+      queue.removeAllListeners(`task_finish`)
```

Comment:
Can `removeAllListeners` potentially cause problems if someone adds other listeners to the queue elsewhere? I realize that we don't do it now but also see how hard it will be to debug this if do some day. Maybe use a more defensive approach and remove individual listeners only?

Reply:
You're right, that's a risk. Arguably the queue shouldn't be reused and it feels like the develop model should be changed.

I did initially have it such that it explicitly removed only one listener for each. It's a lot more code, less maintainable, and serves the same purpose.

Also, once a queue drains, these methods should not matter anymore.

If anything, it feels like batchJobs should create a queue itself and hand that back. Would solve this whole ordeal.

Anyways. Not sure if it's worth doing an explicit remove instead of the all. I'll ask core.

Reply:
If you think the overhead is negligible (I guess it might be) I think removing independent listeners is a better idea even though I completely agree that someone else adding listeners to this would be bad as it stands.

Reply:
Practically we shouldn't have a problem, because nobody is supposed to directly listen to that queue. We have a bit of a weird abstraction problem where it's passed around, while practically it should just pass out a closure over processBatch. I think this is safe to merge.